src/System.Linq/src/System/Linq/ToCollection.cs

```diff
@@ -30,6 +30,22 @@ public static List<TSource> ToList<TSource>(this IEnumerable<TSource> source)
             return source is IIListProvider<TSource> listProvider ? listProvider.ToList() : new List<TSource>(source);
         }
 
+        public static List<TSource> ToList<TSource>(this IEnumerable<TSource> source, int capacity)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+            if (capacity < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity), "Non-negative number required.");
```

Comment:
There's an `Error.ArgumentOutOfRange(string)` method that can be used here. (Assuming negative capacities shouldn't just be ignored).